source/adiabatic_conditions/initial_profile.cc

```diff
@@ -105,7 +110,9 @@ namespace aspect
                          + density * gravity * delta_z;
           temperatures[i] = temperatures[i-1] * (1 +
                                                  alpha * gravity * delta_z * one_over_cp);
+          densities[i-1] = density;
         }
+      densities[n_points-1] = densities[n_points-2];
```

Comment:
I don't think we can compute the density like this. If we are going to use the density from the adiabatic conditions in our computations (as reference density), the profile should be correct. Even if we let `adiabatic_conditions.temperature()` output some constant value when it is not yet initialized (like 0 or the adiabatic surface temperature) the material model would not compute the correct density. I think what we have to do here is something like adding a new function to the interface of `AdiabaticConditions` (something like `initialize_density()`), which is called after `initialize()`, and computes the density profile. We would have to loop over all points again, but we would get the correct reference density profile for all cases.